engine/src/state_chain_observer/client/mod.rs

```diff
@@ -753,26 +756,30 @@ impl SignedExtrinsicClientBuilderTrait for SignedExtrinsicClientBuilder {
 					recorded_version, this_version
 				);
 
-				subxt_client
-					.tx()
-					.sign_and_submit_then_watch_default(
-						&subxt::dynamic::tx(
-							"Validator",
-							"cfe_version",
-							vec![(
-								"new_version",
-								vec![
-									("major", this_version.major),
-									("minor", this_version.minor),
-									("patch", this_version.patch),
-								],
-							)],
-						),
-						&subxt_signer,
-					)
-					.await?
-					.wait_for_in_block()
-					.await?;
+				tokio::time::timeout(
```

Comment:
Just one thing can you add a comment, mentioning the issue? Of course in the end I agree we should add timeouts everywhere. But just while this is a one off timeout.

Reply:
Are you sure it is the second await? Because if the first await on line 778 is the problem, then this timeout will not work.

Reply:
You could put the whole request is an async block and put the timeout around that whole thing?